In [1]:
# imports
import pandas as pd
import sys

In [ ]:
# import custom libraries and scripts
# sys.path.append(os.path.abspath(os.path.join("../..")))
sys.path.append(".")
sys.path.append("..")

from defaults import *
from extract_dataframe import read_json
from extract_dataframe import TweetDfExtractor
from clean_tweets_dataframe import Clean_Tweets

In [ ]:
# read data
global_data = pd.read_json('../data/processed_global_tweet_data.json')
global_data.head(10)

source original_text  polarity  \
0  <a href="http://twitter.com/download/android" ...            NA         0   
1  <a href="http://twitter.com/download/android" ...            NA         0   
2  <a href="http://twitter.com/download/android" ...            NA         0   
3  <a href="http://twitter.com/download/android" ...            NA         0   
4  <a href="http://twitter.com/download/iphone" r...            NA         0   
5  <a href="http://twitter.com/download/android" ...            NA         0   
6  <a href="http://twitter.com/download/android" ...            NA         0   
7  <a href="http://twitter.com/download/android" ...            NA         0   
8  <a href="http://twitter.com/download/android" ...            NA         0   
9  <a href="https://mobile.twitter.com" rel="nofo...            NA         0   

   subjectivity lang  favorite_count  status_count  retweet_count  \
0             0   en               4          8097              2   
1             0   en             691          5831            201   
2             0   en               0          1627              0   
3             0   en               0          1627              0   
4             0   en            1521         18958            381   
5             0   en             116         48483             36   
6             0   en            1106          4173            411   
7             0   en              10         24102              5   
8             0   en            1221           630            239   
9             0   en              49        107188             25   

       screen_name  original_author  followers_count  friends_count  \
0        i_ameztoy                0            20497           2621   
1            ZIisq                1               65            272   
2        Fin21Free                2               85            392   
3        Fin21Free                3               85            392   
4   VizziniDolores                4              910           2608   
5  GraceCh15554845                5              207             54   
6      Philipkuma1                6               12            264   
7        nhohn2011                7              870            508   
8  ClaudioColomaRI                8              127            263   
9        jmarzola1                9              213            877   

   possibly_sensitive                                           hashtags  \
0                 NaN          [{'text': 'City', 'indices': [132, 137]}]   
1                 NaN  [{'text': 'China', 'indices': [18, 24]}, {'tex...   
2                 NaN     [{'text': 'XiJinping', 'indices': [127, 137]}]   
3                 NaN        [{'text': 'XiJinping', 'indices': [9, 19]}]   
4                 NaN                                                 []   
5                 0.0          [{'text': 'Taiwan', 'indices': [84, 91]}]   
6                 NaN  [{'text': 'Taiwan', 'indices': [101, 108]}, {'...   
7                 NaN  [{'text': 'PinkFloyd', 'indices': [18, 28]}, {...   
8                 NaN          [{'text': 'Taiwan', 'indices': [80, 87]}]   
9                 NaN  [{'text': 'USA', 'indices': [66, 70]}, {'text'...   

                                       user_mentions                place  
0  [{'screen_name': 'i_ameztoy', 'name': 'Iban Am...                       
1  [{'screen_name': 'IndoPac_Info', 'name': 'Indo...                       
2  [{'screen_name': 'ZelenskyyUa', 'name': 'Волод...          Netherlands  
3                                                 []          Netherlands  
4  [{'screen_name': 'ChinaUncensored', 'name': 'C...       Ayent, Schweiz  
5  [{'screen_name': 'benedictrogers', 'name': 'Be...  Melbourne, Victoria  
6  [{'screen_name': 'TGTM_Official', 'name': 'The...                       
7  [{'screen_name': 'ChinaInfo777', 'name': 'Chin...         Florida, USA  
8  [{'screen_name': 'AmbQinGang', 'name': 'Qin Ga...  El mundo periférico  
9  [{'screen_name': 'CGMei

In [ ]:
global_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 22000 entries, 0 to 21999
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   source              22000 non-null  object 
 1   original_text       22000 non-null  object 
 2   polarity            22000 non-null  int64  
 3   subjectivity        22000 non-null  int64  
 4   lang                22000 non-null  object 
 5   favorite_count      22000 non-null  int64  
 6   status_count        22000 non-null  int64  
 7   retweet_count       22000 non-null  int64  
 8   screen_name         22000 non-null  object 
 9   original_author     22000 non-null  int64  
 10  followers_count     22000 non-null  int64  
 11  friends_count       22000 non-null  int64  
 12  possibly_sensitive  6191 non-null   float64
 13  hashtags            22000 non-null  object 
 14  user_mentions       22000 non-null  object 
 15  place               22000 non-null  object 
dtypes: f

In [ ]:
global_data.describe()

polarity  subjectivity  favorite_count  status_count  retweet_count  \
count   22000.0       22000.0    22000.000000  2.200000e+04   22000.000000   
mean        0.0           0.0      852.137318  5.446036e+04     176.750182   
std         0.0           0.0     3106.077645  1.454120e+05     498.435765   
min         0.0           0.0        0.000000  1.000000e+00       0.000000   
25%         0.0           0.0        2.000000  2.105750e+03       2.000000   
50%         0.0           0.0      115.000000  1.038750e+04      38.000000   
75%         0.0           0.0      655.000000  4.526150e+04     187.000000   
max         0.0           0.0    65170.000000  4.108317e+06   17409.000000   

       original_author  followers_count  friends_count  possibly_sensitive  
count     22000.000000     2.200000e+04   22000.000000         6191.000000  
mean      10999.500000     1.796764e+04    1563.114455            0.037151  
std        6350.997297     3.030478e+05    4358.651264            0.189146  
min           0.000000     0.000000e+00       0.000000            0.000000  
25%        5499.750000     5.700000e+01     137.000000            0.000000  
50%       10999.500000     2.840000e+02     487.000000            0.000000  
75%       16499.250000     1.324500e+03    1599.000000            0.000000  
max       21999.000000     1.449852e+07  208360.000000            1.000000

## EDA

### Remove duplicated rows

In [ ]:
clean_tweets = Clean_Tweets(global_data)

Automation in Action...!!!


In [ ]:
clean_tweets = clean_tweets.drop_duplicate(global_data)
clean_tweets.shape

TypeError: unhashable type: 'list'

### Remove tweets that are not english

### Missing values

In [ ]:
global_data.possibly_sensitive.value_counts()

: 

As we can see there are 5,961 not sensitive and 230 sensitive tweets
And only 6,191 out of 22,000 tweets are recorded for sensitivity

### Handle missing values

### Remove unwanted characters


### Replace empty strings